xbmc/platform/android/CPUInfoAndroid.h

```diff
@@ -8,17 +8,26 @@
 
 #pragma once
 
+#include "utils/CPUInfo.h"
 #include "utils/Temperature.h"
 
-#include "platform/posix/CPUInfoPosix.h"
+#include "platform/posix/filesystem/PosixFile.h"
 
-class CCPUInfoAndroid : public CCPUInfoPosix
+#include <memory>
+
+using namespace XFILE;
+
+class CCPUInfoAndroid : public CCPUInfo
 {
 public:
   CCPUInfoAndroid();
   ~CCPUInfoAndroid() = default;
 
   bool SupportsCPUUsage() const override { return false; }
   int GetUsedPercentage() override { return 0; }
-  float GetCPUFrequency() override { return 0; }
+  float GetCPUFrequency() override;
+  bool GetTemperature(CTemperature& temperature) override;
```

Comment:
By overriding this you are actually removing functionality. You should passthrough to the CCPUInfoPosix method from this method if you want to do this. See the other comment I left.

Reply:
Yes, -1ºC feature are removed (on purpose) ;-)

Reply:
Then it should be fixed there to be invalid if -1 is set.

Reply:
We do not need to fix (in this PR) a code that is not used in Android / this PR.

Reply:
If you fix it then you won't even need some of the changes you made. If the whole point is to fix the -1 then just do it properly and you will fix it for all POSIX platforms.

Reply:
There is no point in using class `CCPUInfoPosix` in android since this class only offers to read the temperature with a script that is not possible to use in Android. So it doesn't provides any useful functionality for Android.

Reply:
Your fix might be breaking someone else's feature.

Reply:
Sorry, I don't agree